tensorboard/webapp/metrics/views/card_renderer/scalar_card_container.ts

```diff
@@ -426,6 +411,25 @@ export class ScalarCardContainer implements CardRenderer, OnInit, OnDestroy {
       )
     );
 
+    this.columnHeaders$ = combineLatest([
+      this.smoothingEnabled$,
+      this.linkedTimeSelection$,
+    ]).pipe(
+      map(([smoothingEnabled, timeSelection]) => {
+        const headers = [ColumnHeaders.RUN, ColumnHeaders.STEP];
```

Comment:
I think it would be clearer and easier to reason out if the code were a bit more verbose and repetitive:

```
const headers = [];
if (timeSelection === null || timeSelection.endStep == = null) {
  // No step selected or single step selected.
  headers.push(ColumnHeaders.RUN);
  if (smoothingEnabled) {
    headers.push(ColumnHeaders.SMOOTHED);
  }
  headers.push(ColumnHeaders.VALUE);
  headers.push(ColumnHeaders.STEP);
  headers.push(ColumnHeaders.RELATIVE_TIME);
} else {
  // Range selected.
  headers.push(ColumnHeaders.RUN);
  headers.push(ColumnHeaders.VALUE_CHANGE);
  headers.push(ColumnHeaders.STEP);
}
return headers;
```

Reply:
You are right. I was getting too fancy. Done.